cypress/integration/pages/invite-employee/invite-employee.spec.js

```diff
@@ -0,0 +1,33 @@
+describe('Invite new employee', () => {
+  const manager = 'manager@example.com';
+  const password = '123456';
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.get('#email').type(manager);
+    cy.get('#password').type(password);
+    cy.get('[type=submit]').click();
+    cy.contains('Convidar novo empregado').click();
+  });
+
+  it('Invite new employee option must be visible when user is a manager', () => {
+    cy.contains('Convidar novo empregado').should('be.visible');
+  });
+
+  it('should display an e-mail form when visiting invite new employee link', () => {
+    cy.get('#email').should('be.visible');
+    cy.get('button[aria-label="Enviar e-mail"]').should('be.visible');
+  });
+
+  it('should display error message when trying to submit with empty input', () => {
+    cy.get('button[aria-label="Enviar e-mail"]').click();
+    cy.contains('Campo obrigatório').should('be.visible');
+  });
+});
+
+describe('Invite new employee option must not exist', () => {
+  it('Invite new employee option must not exist when user is other than a manager', () => {
+    cy.login('admin@example.com', '123456');
+    cy.contains('Convidar novo empregado').should('not.exist');
+  });
+});
```

Comment:
I feel like this `describe` should be inside of the "Invite new employee" `describe`.

Also, I think the `it` and the `describe` messages could be improvided/simplified, something like:
```suggestion
describe('when the user is not a manage', () => {
  it('Invite new employee option must not exists', () => {
    cy.login('admin@example.com', '123456');
    cy.contains('Convidar novo empregado').should('not.exist');
  });
});
```

Reply:
I agree with you and I tried implementing this way, but I actually split them in two `describe` blocks because in the second one (every role other than a manager) I don't want the `beforeEach()` block to run because I'm logging as an admin and not as a manager. I could've not used the `beforeEach()` block and then not split them in two `describe` blocks, but then I would need to repeat the login steps in each test (`it`).

Yes, the tests titles you're recommending are briefer and easier to read. Thanks.

Reply:
In this case, you can add one more `describe` to separate the manager tests and the 'not a manager' tests, then you use the `beforeEach` inside that describe of `when the user is a manage`, something like this:

![image](https://user-images.githubusercontent.com/101819652/170542861-648a79a4-f000-4a7d-8aea-50766ab678da.png)

Would that make sense?

Reply:
In this case, you can add one more `describe` to separate the manager tests and the 'not a manager' tests, then you use the `beforeEach` inside that describe of `when the user is a manage`, something like this:

![image](https://user-images.githubusercontent.com/101819652/170542861-648a79a4-f000-4a7d-8aea-50766ab678da.png)

Would that make sense?

Reply:
I'm not used to nest `describe` blocks, but that's an approach. Thank you!

Reply:
Sure!

Also, there is this better specs site https://www.betterspecs.org/, where they list some good practices and some approaches you can use while testing something, that might help you with some examples.

> Better Specs focus on Rails testing, but our goal is to create testing guidelines covering most languages and frameworks out there (e.g. Scala, Elixir, React)